lib/ssl/src/inet_tls_dist.erl

```diff
@@ -591,6 +591,10 @@ do_setup_connect(Driver, Kernel, Node, Address, Ip, TcpPort, Version, Type, MyNo
         net_kernel:connecttime()) of
     {ok, #sslsocket{pid = [_, DistCtrl| _]} = SslSocket} ->
             _ = monitor_pid(DistCtrl),
+            case application:get_env(kernel, cb_dist_module) of
+                {ok, CB_Dist_Module} -> CB_Dist_Module:post_setup(?MODULE, SslSocket);
```

Comment:
Passing ?MODULE as a currently used dist module is incorrect.
Both TLS dist modules (inet_tls_dist and inet6_tls_dist) use this function to connect. So technically in case of ipv6 you will pass wrong module here. 

I suggest we don't pass the module at all (for simplicity). We can just assume if this function is called, encryption is on (or was on a moment ago). Hopefully we don't need to do anything like that for non tls connections.
We can also rename the callback to post_tls_setup or something similar for clarity.

Reply:
Oh that makes sense. I've changed it to directly provide the post_tls_setup function instead of the module. Seemed unnecessary to be providing the whole module when just that function will do, and it would allow changing the function name in the future if we wanted, without changing the erlang dependency again.